demos/resource_list.hpp

```diff
@@ -27,6 +28,8 @@ struct resource_list
   std::optional<hal::serial*> console;
   std::optional<hal::steady_clock*> clock;
   std::optional<hal::output_pin*> status_led;
+  std::optional<hal::spi*> spi = std::nullopt;
+  std::optional<hal::output_pin*> spi_chip_select = std::nullopt;
```

Comment:
```suggestion
  std::optional<hal::spi*> spi;
  std::optional<hal::output_pin*> spi_chip_select;
```

Its actually okay to not set these. They default to being nullopt.